### PR TITLE
Moving files over to new repository

### LINKED ISSUE #1: Need to migrate from private repo to public repo

### PR DESCRIPTION
**Pull request by jwendl**

### CODE REVIEW

File: PostGreKeyRotation/Deployment/helm/helmgensecret.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+kubectl create secret docker-registry acrsecret --docker-server jwrotateacr.azurecr.io --docker-email juswen@microsoft.com --docker-username jwrotateacr --docker-password <acr-password> --dry-run=true --output yaml
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/helm/helmgensecret.sh]: Should we sanitize the --docker-server value, email and username?

File: PostGreKeyRotation/Deployment/docker/launchapp.cmd [hunk 2]
```diff
@@ -0,0 +1,2 @@
+start "" "http://localhost:8080/ReadFlexVolume"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/docker/launchapp.cmd]: Still needed?

File: PostGreKeyRotation/Deployment/helm/templates/secret.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+data:
+  .dockerconfigjson: eyJhdXRo...
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/helm/templates/secret.yaml:3]: looks like a truncated value, perhaps replace with something like <GET THIS FROM YOUR DOCKER CONFIG> or the like?

File: PostGreKeyRotation/Deployment/helm/values.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+azure:
+  resourceGroupName: KeyRotation
+  keyVaultName: jwrotatekeyvault
+  applicationGatewayName: jwrotateag
+  subscriptionId: <subscription-id>
+  tenantId: <tenant-id>
+
+connectionString:
+  template: Server=ibpostgreserver.postgres.database.azure.com;Database=ibpostgredb;Port=5432;User Id={RoleName}@ibpostgreserver;Password={Pwd};Ssl Mode=Require;
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/helm/values.yaml]: Sanitize server name, database name?

File: PostGreKeyRotation/Deployment/helm/values.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+container:
+  name: postgrekeyrotation
+  pullPolicy: IfNotPresent
+  image: jwrotateacr.azurecr.io/postgrekeyrotation
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/helm/values.yaml]: Sanitize acr name?

File: PostGreKeyRotation/Deployment/pg-keyrotation-ci.yaml [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    displayName: Deploy Ingress Helm Chart using Shell
+    inputs:
+      failOnStandardError: true
+      azureSubscription: '<service-principal-name>'
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: |
+        helm repo add application-gateway-kubernetes-ingress https://appgwingress.blob.core.windows.net/ingress-azure-helm-package/
+        helm repo update
+        aksfqdn="$(az aks show --resource-group KeyRotation --name jwrotatecluster --query fqdn --output tsv)"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/pg-keyrotation-ci.yaml:134]: sanitize cluster name?

File: PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+export PGPASSWORD="MyB@dP2\$\$wd"
+PGSERVER=${PGSERVER:-'jwpostgreserver.postgres.database.azure.com'}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh:4]: sanitize server name?

File: PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+PGDB=${PGDB:-'jwpostgredb'}
+PGUSER=${PGUSER:-'psqladminun@jwpostgreserver'}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh:6]: sanitize server name?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+BUILDTAG=$2
+
+if [ -z "${ACTIVEROLE}" ]; then
+    echo "Usage:  $0 <active role> <build tag>"
+    exit 1;
+else
+    echo "ACTIVEROLE: ${ACTIVEROLE}"
+    echo "BUILDTAG: ${BUILDTAG}"
+
+    DBHOSTNAME="jwpostgreserver"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:13]: sanitize server name?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    DBNAME="jwpostgredb"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:14]: sanitize db name?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    --set productionSlot=${PRODUCTIONSLOT} \
+    --set blue.enabled=${BLUEENABLED} \
+    --set blue.tag=${BUILDTAG} \
+    --set blue.connectionString.roleName="pgapproleblue" \
+    --set blue.connectionString.mountPoint="/mnt/keyvault/pgmainbluepwd" \
+    --set green.enabled=${GREENENABLED}  \
+    --set green.tag=${BUILDTAG} \
+    --set green.connectionString.roleName="pgapprolegreen" \
+    --set green.connectionString.mountPoint="/mnt/keyvault/pgmaingreenpwd" \
+    --set connectionString.template="Server=${DBHOSTNAME}.postgres.database.azure.com;Database=${DBNAME};Port=5432;User Id={RoleName}@${DBHOSTNAME};Password={Pwd};Ssl Mode=Require;"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:45]: Server, db name?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-staging.sh [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    --set productionSlot=${ACTIVEROLE} \
+    --set blue.enabled=true \
+    --set blue.tag=${BUILDTAG} \
+    --set blue.connectionString.roleName="pgapproleblue" \
+    --set blue.connectionString.mountPoint="/mnt/keyvault/pgmainbluepwd" \
+    --set green.enabled=true \
+    --set green.tag=${BUILDTAG} \
+    --set green.connectionString.roleName="pgapprolegreen" \
+    --set green.connectionString.mountPoint="/mnt/keyvault/pgmaingreenpwd" \
+    --set connectionString.template="Server=${DBHOSTNAME}.postgres.database.azure.com;Database=${DBNAME};Port=5432;User Id={RoleName}@${DBHOSTNAME};Password={Pwd};Ssl Mode=Require;"
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-deploy-staging.sh:32]: server, db name

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh [hunk 13]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+PGSERVER=${PGSERVER:-"jwpostgreserver.postgres.database.azure.com"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:3]: server name

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+PGDB=${PGDB:-"jwpostgredb"}
+PGUSER=${PGUSER:-"psqladminun@jwpostgreserver"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:5]: server name

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+ACTIVEROLE=$1
+
+if [ -z "${ACTIVEROLE}" ]; then
+    echo "Usage:  $0 <active role>"
+    exit 1;
+else
+    echo "ACTIVEROLE: ${ACTIVEROLE}"
+
+    PGSERVER=${PGSERVER:-"jwpostgreserver.postgres.database.azure.com"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:11]: server name

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+    PGDB=${PGDB:-"jwpostgredb"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:12]: db name

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+    PGUSER=${PGUSER:-"psqladminun@jwpostgreserver"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:13]: server name

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh [hunk 18]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+PGSERVER=${PGSERVER:-"jwpostgreserver.postgres.database.azure.com"}
+PGDB=${PGDB:-"jwpostgredb"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:4]: db name

File: PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+
+ACTIVEROLE=$1
+
+if [ -z "${ACTIVEROLE}" ]; then
+    echo "Usage:  $0 <active role>"
+    exit 1;
+else
+    echo "ACTIVEROLE: ${ACTIVEROLE}"
+
+    PGSERVER=${PGSERVER:-"jwpostgreserver.postgres.database.azure.com"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh:11]: server name

File: PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+    PGDB=${PGDB:-"jwpostgredb"}
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh:12]: db name

File: PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh [hunk 21]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+existingBlueValue=$(az keyvault secret show --id https://jwrotatekeyvault.vault.azure.net/secrets/pgmainbluepwd --vault-name jwrotatekeyvault)
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh:3]: keyvault uri

File: PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+if [ -z "$existingBlueValue" ]
+then
+    echo "Secret pgmainbluepwd does not exist, we will create it..."
+    az keyvault secret set --name pgmainbluepwd --vault-name jwrotatekeyvault --value "MyB@dP2\$\$wd"
+else
+    echo "Secret pgmainbluepwd already exists"
+fi
+
+existingGreenValue=$(az keyvault secret show --id https://jwrotatekeyvault.vault.azure.net/secrets/pgmaingreenpwd --vault-name jwrotatekeyvault)
```
Comment (ianisms) [inline PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh:12]: keyvault uri

File: PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLConnectResult.cs [hunk 23]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace PostGreKeyRotation.Models
+{
+    public class PostgreSQLConnectResult
```
Comment (ianisms) [inline PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLConnectResult.cs]: Not to nitpick but perhaps this file should be my commit?

File: PostGreKeyRotation/PostGreKeyRotation/Controllers/PostgreSQLController.cs [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace PostGreKeyRotation.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class PostgreSQLController
```
Comment (ianisms) [inline PostGreKeyRotation/PostGreKeyRotation/Controllers/PostgreSQLController.cs]: Not to nitpick but perhaps this file should be my commit?

File: PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLControllerConfig.cs [hunk 25]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PostGreKeyRotation.Models
+{
+    public class PostgreSQLControllerConfig
```
Comment (ianisms) [inline PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLControllerConfig.cs]: Not to nitpick but perhaps this file should be my commit?

File: PostGreKeyRotation/PostGreKeyRotation/Models/SimpleUser.cs [hunk 26]
```diff
@@ -0,0 +1,9 @@
+﻿namespace PostGreKeyRotation.Models
+{
+    public class SimpleUser
```
Comment (ianisms) [inline PostGreKeyRotation/PostGreKeyRotation/Models/SimpleUser.cs]: Not to nitpick but perhaps this file should be my commit?

File: PostGreKeyRotation/PostGreKeyRotation/appsettings.json [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  },
+  "AllowedHosts": "*",
+  "PostgreSQLControllerConfig": {
+    "ConnectionStringRoleName": "pgapproleblue",
+    "ConnectionStringPwdMountPoint": "/mnt/keyvault/pgmainbluepwd",
+    "ConnectionStringTemplate": "Server=jwpostgreserver.postgres.database.azure.com;Database=jwpostgredb;Port=5432;User Id={RoleName}@jwpostgreserver;Password={Pwd};Ssl Mode=Require;"
```
Comment (ianisms) [inline PostGreKeyRotation/PostGreKeyRotation/appsettings.json]: server and db name

**Review by ianisms — CHANGES_REQUESTED** [review]
Looks good, great job!  A few comments peppered throughout.

Summary:
- Should we sanitize the ACR uri, db server name, db name?
- Not to nitpick and hate to complicate, shall I be the one to add the PostgreSQLContoller bits to the app?  Would mean that the app would not function until I commit.

File: PostGreKeyRotation/Deployment/helm/helmgensecret.sh (hunk 1 above)
Comment (jwendl) [reply PostGreKeyRotation/Deployment/helm/helmgensecret.sh]: Ok fixed this.

File: PostGreKeyRotation/Deployment/helm/templates/secret.yaml (hunk 3 above)
Comment (jwendl) [reply PostGreKeyRotation/Deployment/helm/templates/secret.yaml:3]: Let's leave it for now, just not sure what to put here anyways.

File: PostGreKeyRotation/Deployment/helm/values.yaml (hunk 4 above)
Comment (jwendl) [reply PostGreKeyRotation/Deployment/helm/values.yaml]: Fixed

File: PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLConnectResult.cs (hunk 23 above)
Comment (jwendl) [reply PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLConnectResult.cs]: Fixed

File: PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLControllerConfig.cs (hunk 25 above)
Comment (jwendl) [reply PostGreKeyRotation/PostGreKeyRotation/Models/PostgreSQLControllerConfig.cs]: Fixed

File: PostGreKeyRotation/PostGreKeyRotation/Controllers/PostgreSQLController.cs (hunk 24 above)
Comment (jwendl) [reply PostGreKeyRotation/PostGreKeyRotation/Controllers/PostgreSQLController.cs]: Fixed

File: PostGreKeyRotation/PostGreKeyRotation/Models/SimpleUser.cs (hunk 26 above)
Comment (jwendl) [reply PostGreKeyRotation/PostGreKeyRotation/Models/SimpleUser.cs]: Fixed

File: PostGreKeyRotation/PostGreKeyRotation/appsettings.json (hunk 27 above)
Comment (jwendl) [reply PostGreKeyRotation/PostGreKeyRotation/appsettings.json]: Fixed

File: PostGreKeyRotation/Deployment/docker/launchapp.cmd (hunk 2 above)
Comment (jwendl) [reply PostGreKeyRotation/Deployment/docker/launchapp.cmd]: Fixed

File: PostGreKeyRotation/Deployment/helm/values.yaml (hunk 5 above)
Comment (jwendl) [reply PostGreKeyRotation/Deployment/helm/values.yaml]: Fixed

File: PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh (hunk 7 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh:4]: This one too?

File: PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh (hunk 8 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/deploy-ddl.sh:6]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh (hunk 9 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:13]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh (hunk 10 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:14]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh (hunk 11 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-deploy-production.sh:45]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-deploy-staging.sh (hunk 12 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-deploy-staging.sh:32]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh (hunk 13 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:3]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh (hunk 18 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:4]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh (hunk 14 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-reset-postgresql-roles.sh:5]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh (hunk 15 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:11]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh (hunk 16 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:12]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh (hunk 17 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-rotate-keys.sh:13]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh (hunk 19 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh:11]: This one too?

File: PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh (hunk 20 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/pg-swap-active.sh:12]: This one too?

File: PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh (hunk 21 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh:3]: This one too?

File: PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh (hunk 22 above)
Comment (ianisms) [reply PostGreKeyRotation/Deployment/scripts/set-keyvault-password.sh:12]: This one too?

**Review by ianisms — COMMENTED** [review]
Just a few more ACR uris, db servers and db names...

**Pull request merged**